plugins/widget/plugin.js

```diff
@@ -3245,10 +3282,33 @@
 
 			// Prevent cutting in read-only editor (#1570).
 			if ( isCut && !editor.readOnly ) {
-				widget.repository.del( widget );
-				editor.fire( 'saveSnapshot' );
+				handleCut();
 			}
 		}, 100 ); // Use 100ms, so Chrome (@Mac) will be able to grab the content.
+
+		function getClipboardHtml() {
+			var selectedHtml = editor.getSelectedHtml( true );
+
+			if ( editor.widgets.focused ) {
+				return editor.widgets.focused.getClipboardHtml();
+			}
+
+			editor.once( 'toDataFormat', function( evt ) {
+				evt.data.widgetsCopy = true;
+			}, null, null, -1 );
+
+			return editor.dataProcessor.toDataFormat( selectedHtml );
```

Comment:
This execution operates on the whole selected HTML, which means it also touches non-widget content. It seems for me that such change is a bit out of the scope of the widget plugin. Maybe it should be implemented directly inside `clipboard` plugin?

Although it would require moving whole copybin logic into the clipboard, which maybe not the best idea as it will complicate whole improvement :thinking:

Reply:
The result should be the same as if we would transform only widgets IMO. Moving whole copybin logic to the `clipboard` plugin would be nice, but I fear that it could be too much work.